src/include/gcc.h

```diff
@@ -128,6 +128,16 @@ __wt_atomic_cas_ptr(void *vp, void *old, void *newv)
     return (WT_ATOMIC_CAS((void **)vp, &old, newv));
 }
 
+/*
+ * __wt_atomic_load_acquire64 --
+ *     Read a with atomic acquire synchronization.
+ */
+static inline void
+__wt_atomic_load_acquire64(void *ret, void *ptr)
+{
+    __atomic_load((void **)ptr, (void **)ret, __ATOMIC_ACQUIRE);
```

Comment:
Do you know if this implies a compiler barrier as well as a hardware barrier? i.e: does an `atomic_load` imply the equivalent of a `WT_BARRIER` as well?